app/tests/test_search.py

```diff
@@ -2009,3 +2029,152 @@ def test_search_transferring_body_all_accordions_open_with_open_all_query(
 
         assert "checked" in checkbox.attrs
         assert all("open" in details.attrs for details in details_elements)
+
+    @patch("app.main.util.search_utils.OpenSearch")
+    def test_search_transferring_body_file_name_in_source_shown_if_not_in_highlight(
+        self,
+        mock_search_client,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the search page with a query
+        If the file_name is not found inside highlight
+        Then it should be shown as the value inside _source AND be a link AND be the 2nd row
+        """
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value=os_mock_return_tb
+        )
+        mock_standard_user(client, "first_body")
+
+        transferring_body_id = browse_consignment_files[
+            0
+        ].consignment.series.body.BodyId
+
+        response = client.get(
+            f"{self.route_url}/{transferring_body_id}",
+            data={"query": "test"},
+        )
+
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.data, "html.parser")
+        anchors = soup.find_all("a")
+        anchors_text = []
+        for anchor in anchors:
+            anchors_text.append(anchor.get_text(strip=True))
+        assert expected_file_name in anchors_text
+
+        table_body = soup.find("tbody")
+        table_rows_cell_values = get_table_rows_cell_values(table_body)
+        assert table_rows_cell_values[1] == ["File name", expected_file_name]
+
+    @patch("app.main.routes.uuid.uuid4")
+    @patch("app.main.util.search_utils.OpenSearch")
+    def test_search_transferring_body_highlight_file_name_prioritized_over_source(
+        self,
+        mock_search_client,
+        mock_uuid4,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the search page with a query
+        If highlight has a file_name field
+        Then it should be shown in place of the file_name inside _source AND be an achor AND is the 2nd row
+        """
+        mock_uuid4.return_value.hex = "highlight_tag"
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value={
+                "hits": {
+                    "total": {
+                        "value": 1000,
+                    },
+                    "hits": [
+                        {
+                            "_source": {
+                                "file_name": "fifth_file.doc",
```

Comment:
noticed this test still has the odd mocked data where the filename is different in source and highlight. Can we updated this and the assertions as discussed